adapters/repos/db/index.go

```diff
@@ -123,6 +125,23 @@ func NewIndex(ctx context.Context, config IndexConfig,
 
 		shard, err := NewShard(ctx, promMetrics, shardName, index, class)
 		if err != nil {
+			// replace combined error message with one more user friendly in case it refers
+			// to unsupported newly introduced RoaringSet bucket's strategy
+			var errns lsmkv.ErrorNotSupported
+			if errors.As(err, &errns) && errns.Strategy() == lsmkv.SegmentStrategyRoaringSet {
```

Comment:
if it is not too dificult, would it be possible to add a test for this?

I assume your regex is right, but it's such a pain to verify

Reply:
As this change was only intended for v1.17 (and is removed in 1.18) I wanted to implement it in as effortless way as possible. I can refactor it and add some tests if needed though.